img.js

```diff
@@ -122,8 +122,32 @@ function getValidWidths(originalWidth, widths = [], allowUpscale = false) {
   return filtered.sort((a, b) => a - b);
 }
 
+function getFileHash(src, options) {
+  const hash = createHash("sha1");
+
+  const opts = {
+    sharpOptions: options.sharpOptions,
+    sharpWebpOptions: options.sharpWebpOptions,
+    sharpPngOptions: options.sharpPngOptions,
+    sharpJpegOptions: options.sharpJpegOptions,
+    sharpAvifOptions: options.sharpAvifOptions
+  };
+
+  if(fs.existsSync(src)) {
+    const fileContent = fs.readFileSync(src);
+    hash.update(fileContent);
+  } else {
+    hash.update(src);
+  }
+
+  hash.update(JSON.stringify(opts));
+
+  return hash.digest('hex').substring(0, 8);
```

Comment:
Yeah, it's probably not very collision safe 😅. The substring was to keep the hashes the same length as `short-hash`, but if it's okay, let's have the complete length hash!

Yes, I'll add the method to exports.

Reply:
Ah, I do really like the shorter hashes. This stackexchange post seems eerily relevant:

https://security.stackexchange.com/questions/72673/how-bad-is-it-to-truncate-a-hash

> If there is no attacker and you are just fighting bad luck, then truncating to 8 bytes incurs the risk of spurious collisions; you should reach your first collision after (on average) entering a few billions of entries. Depending on your situation, this may or may not be tolerable.

Will a project have a billion images? 😅

Reply:
Haha yeah! 😂 

But we are actually truncating to just 4 bytes, since 1 hex is just 1/2 a byte. Using some very very non-cryptographic calculations, we might have a collision every 160-240 images or so.

But I think we can encode it to [`base64url`](https://nodejs.org/api/buffer.html#buffer_buffers_and_character_encodings) and that should give us 6 bytes in 8 letters.

Reply:
I think it’s worth it to use a little bit longer hash to avoid a new dep, fwiw

Reply:
Commit b0ee0a9 uses sha256, and has 60 bits (or 10 base64 letters).

Also, I found a table for collision probability for a perfect hash function: https://en.wikipedia.org/wiki/Birthday_problem#Probability_table

If you don't like the look of the new hashes, we can definitely revert to sha1 and hex. The table suggests it should be relatively safe.

Reply:
`base64url` is only available from Node v15 and up. 

There are two userland implementations:
- https://www.npmjs.com/package/base64-url
- https://www.npmjs.com/package/base64url

The latter seemed more trustworthy, so 2f0f27e uses base64url package to encode hashes.